src/perf_counters.h

```diff
@@ -35,6 +35,13 @@
 #pragma warning(disable : 4251)
 #endif
 
+#ifdef BENCHMARK_OS_MACOSX
+  //this is the OSX syscall wrapper function, implemented in libpthread, that returns
+  //per thread instruction and cycle counters. The counts remain correct over context
+  //switches and core migrations.
+  extern "C"  int thread_selfcounts(int type, void *buf, size_t nbytes);
```

Comment:
is this api stable over osx/libpthread version updates?

Reply:
Its been stable for years, it was added in 2017 and is in the source code as syscall 186, its on Intel and Arm versions of MacOS, its technically on the XNU OS's but you'll have a hard time calling it on anything else because of sandboxing.

Reply:
SGTM .. thanks for the complete answer.